libs/libc/pthread/pthread_getcpuclockid.c

```diff
@@ -0,0 +1,64 @@
+/****************************************************************************
+ * libs/libc/pthread/pthread_getcpuclockid.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <pthread.h>
+#include <errno.h>
+#include <time.h>
+#include <nuttx/clock.h>
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: pthread_getcpuclockid
+ *
+ * Description:
+ *   The pthread_getcpuclockid() function using to access a thread CPU-time
+ *   clock
+ *
+ * Input Parameters:
+ *   thread_id - The id of the thread to fetch the running time
+ *   clockid - The clockid with type CLOCK_THREAD_CPUTIME_ID that required
+ *
+ * Returned Value:
+ *    Return 0 on success, return error number on error
+ *
+ ****************************************************************************/
+
+int pthread_getcpuclockid(pthread_t thread_id, FAR clockid_t *clockid)
```

Comment:
@xiaoxiang781216 I would like to pause here and discuss the same question I posted in https://github.com/apache/nuttx/pull/9691 because implementation is very similar to `clock_getcpuclockid()`.
The main question I want to discuss is: Do we target to support compatibility between apps that use `config.h`? The good example is `clock_getres()`. The interface returns the value of `NSEC_PER_TICK` that is based on `CONFIG_USEC_PER_TICK`. If it is mandatory to use the same `config.h` file the `clock_getres()` can be a libc call and we can eliminate 1 system call. If the target to support kernel update and keep old apps and kernel changes `CONFIG_USEC_PER_TICK` then `clock_getres()` has to "ask" kernel for actual value of clock resolution.
I have a feeling that apps and kernel are tight together to use same `config.h`

Reply:
> @xiaoxiang781216 I would like to pause here and discuss the same question I posted in #9691 because implementation is very similar to `clock_getcpuclockid()`. The main question I want to discuss is: Do we target to support compatibility between apps that use `config.h`?

Do you mean the source code compatibility or binary compatibility? I think it's impossible task to achieve the binary compatibility.

 The good example is `clock_getres()`. The interface returns the value of `NSEC_PER_TICK` that is based on `CONFIG_USEC_PER_TICK`. If it is mandatory to use the same `config.h` file the `clock_getres()` can be a libc call and we can eliminate 1 system call. If the target to support kernel update and keep old apps and kernel changes `CONFIG_USEC_PER_TICK` then `clock_getres()` has to "ask" kernel for actual value of clock resolution. I have a feeling that apps and kernel are tight together to use same `config.h`

Yes, actually, we have a work item to move the most stuff from sched/clock and sched/environ/ to libc/,